.github/workflows/publish.yaml

```diff
@@ -0,0 +1,83 @@
+name: Publish Docker image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - apps/**
+  pull_request:
+    paths:
+      - apps/**
+
+jobs:
+  build:
+    permissions:
+      packages: write
+      contents: read
+    name: Build
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        containers:
+          - app: tftpd-hpa
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+
+      - name: Prepare
+        id: prepare
+        shell: bash
+        run: |
+          # Grab the version from the VERSION file
+          VERSION=$(cat ./apps/${{ matrix.containers.app }}/VERSION )
+
+          # Get the dockerhub token
+          TOKEN=$(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ixsystems/${{ matrix.containers.app }}:pull" | jq '.token' --raw-output)
+
+          # Check if the "production" version already exists in dokerhub
+          TAGS=$(curl -H "Authorization: Bearer $TOKEN" "https://index.docker.io/v2/ixsystems/${{ matrix.containers.app }}/tags/list")
+
+          # If we failed to get the tags, fail fast
+          if echo "$TAGS" | grep -q "errors"; then
```

Comment:
If the remote repository has been created,the first push won't go through because there are no tags currently?
We should account for that i think..
So we can have separate logic that checks to ensure remote repository exists and then we check if there are tags or not.
If no tags are found, it should be fine and we can push the image then..we do need to make sure the API call did not fail and it is actually the tags which were not found.

Reply:
Updated the script:
From some testing I did:
Non public repo or non existing repo -> `UNAUTHORIZED`
Existing public repo with no tags -> `NAME_UNKNOWN`

Reply:
Switched to using `docker manifest inspect`.
This should work better.